### PR TITLE
Pipeline, Polina Shamanina - 21FPL1

### PR DESCRIPTION
**Pull request by pdshh**

### CODE REVIEW

File: lab_6_pipeline/pipeline.py [hunk 1]
```diff
@@ -3,9 +3,36 @@
...
+import re
 
-from core_utils.article.article import SentenceProtocol
+from core_utils.article.article import SentenceProtocol, get_article_id_from_filepath
+from core_utils.article.io import from_raw, to_cleaned, to_conllu
 from core_utils.article.ud import OpencorporaTagProtocol, TagConverter
+from core_utils.constants import ASSETS_PATH
+
+
+class FileNotFoundError(Exception):
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: This exception already exists

File: lab_6_pipeline/pipeline.py [hunk 2]
```diff
@@ -18,21 +45,54 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+            raise FileNotFoundError
+
+        if not self.path_to_data.is_dir():
+            raise NotADirectoryError
+
+        if not any(self.path_to_data.iterdir()):
+            raise EmptyDirectoryError
+
+        raw_files = [file for file in self.path_to_data.glob('*_raw.txt')]
+        # meta_files = [file for file in self.path_to_data.glob(r'*_meta.json')]
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: don't forget to remove unused code before handing in

**Review by marina-kaz — COMMENTED** [review]
Good start!

File: lab_6_pipeline/pipeline.py [hunk 3]
```diff
@@ -144,16 +206,27 @@ def __init__(self, corpus_manager: CorpusManager):
...
+        self._corpus = corpus_manager
 
     def _process(self, text: str) -> List[ConlluSentence]:
         """
         Returns the text representation as the list of ConlluSentence
         """
+        conllu_sentences = []
+        sentences = split_by_sentence(text)
+        for position, sentence in enumerate(sentences):
+            tokens = [ConlluToken(word.strip()) for word in sentence.split()]
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: Can you explain why you need to strip `word`?

**Review by marina-kaz — COMMENTED** [review]
It looks good for the mark 4 :) 
Maybe consider aiming for a higher mark?

File: lab_6_pipeline/pipeline.py [hunk 4]
```diff
@@ -18,21 +35,64 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+                raise InconsistentDatasetError
+
+        list_of_raw_ids = []
+        for file in raw_files:
+            list_of_raw_ids.append(int(file.name[:file.name.index('_')]))
+        if sorted(list_of_raw_ids) != list(range(1, len(list_of_raw_ids) + 1)):
+            raise InconsistentDatasetError
+
+        list_of_meta_ids = []
+        for file in raw_files:
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: It looks like you meant to iterate over meta files, but instead iterate over raw ones

File: lab_6_pipeline/pipeline.py [hunk 5]
```diff
@@ -144,16 +255,57 @@ def __init__(self, corpus_manager: CorpusManager):
...
         Returns the text representation as the list of ConlluSentence
         """
+        conllu_sentences = []
+        sentences = split_by_sentence(text)
+        for position, sentence in enumerate(sentences):
+            mystem_sentence = self._mystem.analyze(sentence)
+            conllu_tokens = []
+            token_counter = 0
+            for token in mystem_sentence:
+                if not re.match(r'\w+|[.]', token['text']):
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py:270]: what kind of tokens won't be processed?

**Review by marina-kaz — COMMENTED** [review]
Looks good! A little to fix

File: lab_6_pipeline/pipeline.py (hunk 5 above)
Comment (pdshh) [reply lab_6_pipeline/pipeline.py:270]: The ones that are not words/numbers/full stops

**Pull request closed without merging**